packages/formstr-app/src/containers/CreateFormNew/components/AnswerSettings/RightAnswer.tsx

```diff
@@ -1,44 +1,111 @@
 import { AnswerSettings, AnswerTypes } from "@formstr/sdk/dist/interfaces";
 import { InputFiller } from "../../../../old/containers/FormFiller/QuestionNode/InputFiller";
-import { Tooltip, Typography } from "antd";
+import { Tooltip, Typography, Select } from "antd";
 
 const { Text } = Typography;
+const { Option } = Select;
+
+type ChoiceItem = [string, string];
 
 interface RightAnswerProps {
-  answerType: AnswerTypes;
-  answerSettings: AnswerSettings;
-  onChange: (answer: string) => void;
+  answerType: AnswerTypes | string;
+  answerSettings: AnswerSettings & {
+    renderElement?: string;
+    choices?: string;
+  };
+  onChange: (answer: string | string[]) => void;
 }
 
 export const RightAnswer: React.FC<RightAnswerProps> = ({
   answerType,
   answerSettings,
   onChange,
 }) => {
-  function convertV1toV2(answerSettings: AnswerSettings) {
-    const newAnswerSettings = {
-      ...answerSettings,
-      choices: answerSettings?.choices?.map((choice) => {
-        return {
-          ...choice,
-          choiceId: choice.choiceId!,
-        };
-      }),
-    };
-    return newAnswerSettings;
-  }
+  const getChoices = (): ChoiceItem[] => {
+    try {
+      if (typeof answerSettings.choices === 'string') {
+        return JSON.parse(answerSettings.choices);
+      }
+      return [];
+    } catch (e) {
+      console.error('Error parsing choices:', e);
+      return [];
+    }
+  };
+
+  const renderAnswerInput = () => {
+    const currentAnswer = answerSettings?.validationRules?.match?.answer;
+    const renderElement = answerSettings.renderElement;
+    
+    if (renderElement === 'checkboxes') {
```

Comment:
you shouldn't have to do it, the inputFiller should already solve for different input types right?